### DISCUSSION TITLE
Edit `helm-occur` results with wgrep

### DISCUSSION THREAD

**Discussion by ipocentro87** [emacs-helm/helm#2665] [Q&A] [answered]
Hi all, 
I was wondering how to edit the saved buffer generated by `helm-occur` with `wgrep`, similarly to what I can do with `helm-do-grep-ag`.
So when I type `helm-occur` I select the action "Save buffer", that creates me a `helm-occur` buffer. 
I expected that after `wgrep-change-to-wgrep-mode`  + changes + `wgrep-finish-edit` I was able to modify the occurrences in the buffer, but the function returns with "(No changes to be performed)".
Do you have any clue on how I can apply the changes?
Thanks in advance

**Comment by thierryvolpiatto** [comment]
Marcello Ranone ***@***.***> writes:

> Hi all,
> I was wondering how to edit the saved buffer generated by helm-occur with wgrep, similarly to what I can do with helm-do-grep-ag.
> So when I type helm-occur I select the action "Save buffer", that creates me a helm-occur buffer.
> I expected that after wgrep-change-to-wgrep-mode + changes + wgrep-finish-edit I was able to modify the occurrences in the buffer,
> but the function returns with "(No changes to be performed)".
> Do you have any clue on how I can apply the changes?

I guess you don't have installed and or loaded wgrep-helm.el, here I
install wgrep from source (git) and the file is included but IIRC there
is a package providing this file.
With this installed it is working fine here. 

> 
> Thanks in advance
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.

-- 
Thierry

**Reply by ipocentro87** [reply]
Hi Thierry, I already have wgrep-helm (version 20230202.315) installed in Emacs, and `(wgrep-helm-setup)` is in `helm-occur-mode-hook`.

**Reply by ipocentro87** [reply, marked as answer]
I've found why it doesn't work on my side. I had changed the `helm-occur-transformer` function to display results in this fashion `%4s %s` instead of `%s:%s` to have line numbers aligned.
By reverting the change it works as expected. I guess wgrep needs occurrences to be in the form `line_number:file`.

**Comment by thierryvolpiatto** [comment]
Marcello Ranone ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
> I've found why it doesn't work on my side. I had changed the helm-occur-transformer function to display results in this fashion %4s %s
> instead of %s:%s to have line numbers aligned.
> By reverting the change it works as expected. I guess wgrep needs occurrences to be in the form line_number:file.

Not exactly, wgrep needs file:linum:xxxxxxx or linum:xxxxxxx where linum
contains a text property handling file or buffer name (helm grep or helm
occur).

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.

-- 
Thierry
